zkevm-circuits/src/witness/tx.rs

```diff
@@ -307,52 +318,448 @@ impl Transaction {
             })
             .collect()
     }
-}
 
-impl Encodable for Transaction {
-    fn rlp_append(&self, s: &mut RlpStream) {
-        s.begin_list(9);
-        s.append(&Word::from(self.nonce));
-        s.append(&self.gas_price);
-        s.append(&Word::from(self.gas));
-        if let Some(addr) = self.callee_address {
-            s.append(&addr);
+    pub(crate) fn gen_rlp_witness<F: Field>(
+        &self,
+        is_hash: bool,
+        challenges: &Challenges<Value<F>>,
+    ) -> Vec<RlpFsmWitnessRow<F>> {
+        let (rlp_bytes, format) = if is_hash {
+            (
+                self.rlp_signed.clone(),
+                match self.tx_type {
+                    TxTypes::Eip155 => TxHashEip155,
+                    TxTypes::PreEip155 => TxHashPreEip155,
+                    TxTypes::Eip1559 => TxHashEip1559,
+                    TxTypes::L1Msg => L1MsgHash,
+                    _ => unimplemented!(),
```

Comment:
nitpick: might be better to replace with a reason such as
```suggestion
                    tx_type => unreachable!("tx type: {:?} not supported", tx_type),
```

Reply:
fixed in https://github.com/scroll-tech/zkevm-circuits/pull/488/commits/f3f656e20de67c5028dbb1ce1ea0ced30b35c6b2